### PR TITLE
Fix check of filename length

### PR DESCRIPTION
**Pull request by brondani**
Fix check of filename length in `RealDiskInterface::Stat` when long paths support in Windows is enabled.
Add the application manifest to the `ninja_test` executable target to enable long paths support during tests.
Add the test case `DiskInterfaceTest.StatExistingFileWithLongPath`.
Address https://github.com/ninja-build/ninja/issues/2266 (`Filename longer than 260 characters` error).

### CODE REVIEW

File: src/disk_interface.cc [hunk 1]
```diff
@@ -157,13 +158,27 @@ bool DiskInterface::MakeDirs(const string& path) {
 }
 
 // RealDiskInterface -----------------------------------------------------------
+RealDiskInterface::RealDiskInterface() 
+#ifdef _WIN32
+: use_cache_(false), long_paths_enabled_(false) {
+  setlocale(LC_ALL, "");
```
Comment (orgads) [inline src/disk_interface.cc:164]: How is this related? It causes a regression. See #2435.

Comment (brondani) [reply src/disk_interface.cc:164]: It seems to be a leftover, thanks for reporting!
